src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,235 @@
+const React = require('react');
+const T = require('prop-types');
+const NavLink = require('react-router-dom').NavLink;
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+const { Button, TextField, FormHelperText, Divider, Typography } = require('@material-ui/core');
+const { FormWrapper, TextWrapper, ButtonWrapper } = require('styles/global-components.js');
+
+
+module.exports = class Settings extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        onSubmit: T.func.isRequired,
+        errorMessage: T.string,
+        userDetails: T.shape({
+            firstName: T.string.isRequired,
+            lastName: T.string.isRequired,
+            email: T.string.isRequired
+        })
+    };
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            firstName: props.userDetails.firstName || '',
+            lastName: props.userDetails.lastName || '',
+            email: props.userDetails.email || '',
+            currentPassword: '',
+            password: '',
+            confirmPassword: '',
+            isBlurred: {
+                email: false,
+                confirmPassword: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['firstName', 'lastName', 'email', 'currentPassword', 'password', 'confirmPassword'],
+            get: (someProps, field) => this.state[field],
+            act: (field, value) => this.setState({ [field]: value }),
+            getFormValue: {
+                '*': this.getFormValue.bind(this)
+            }
+        });
+
+        this.fieldBlurred = this._fieldBlurred.bind(this);
```

Comment:
If all these `bind()`s hurt you, I think we (and the greater community) have become more open to starting with arrows,
```js
    onBlur={() => this.fieldBlurred()}
```

Either is cool though!

Reply:
I think it's alright to have both.  It's new-ish that this is allowed by our react linting rules, so I expect we'll be used to seeing/understanding both.